### DISCUSSION TITLE
OpenDTU Kommunikationsproblem bei 2 Wechselrichtern

### DISCUSSION THREAD

**Discussion by JGebauer1** [tbnobody/OpenDTU#1813] [Q&A]
Bis gestern lief die DTU super mit zwei WR, 1 HM 1500, einem HM 800. Beide werden per ioBroker geregelt, der HM 800 übernimmt eine Akkueinspeisung, wird also je nach Bedarf ein/ausgeschaltet und in der Leistung geregelt.
Seit gestern Mittag habe ich endlose Probleme, die ich jetzt einigermaßen analysiert habe: Immer wenn beide WR durch die DTU abgerufen werden, verlieren sie den Empfang. Es scheint, als würden sie Befehle weiterhin empfangen, allerdings stark zeitversetzt.
Wenn ich Empfang/Senden eines der WR ausschalte (oder ihn sonstwie deaktiviere), geht der andere wieder perfekt mit der DTU. Sobald der Empfang/Senden wieder aktiviert wird, dauert es höchstens eine Minute und die Kommunikation ist weg.

Meine DTU ist selbstgebaut mit Steckkabeln. 
- Ich habe versucht, nach Anleitung einen Kondensator an den Stromanschluss des Funkmoduls zu klemmen, hat nichts gebracht.
- Ich habe das Funkmodul gegen ein anderes hier noch herumliegendes getauscht, kein Unterschied.
- Die DTU ist an einem iPad-Netzteil mit 2A Leistung.
- Firmware ist die neueste Generic. Das Modul ist ein ESP32 DEVKITV1
- Das Funkmodul ist mit Antenne
- Ein Display ist auch dran.
Vielleicht ist der ESP einfach defekt?

Falls jemand die Lösung weiß... Danke J.

**Comment by MetaChuh** [comment]
hi @JGebauer1 

kann vieles sein, wenn die installation bei dir bis jetzt normal funktioniert hat.
2 funkmodule mit antenne hast du bereits getestet, und da's mit einem wr einwandfrei funktioniert, und die dtu nicht dauerrebootet, kann man den esp32 als fehlerquelle nahezu ausgliedern.

am besten mal alles abdrehen was evtl. zu viele gleiche befehle oder abfragen für die inverter an die dtu schickt (api adapter, mqtt, cron auf die api) und dann nur das bare metal opendtu web interface zum testen verwenden.
wenn dies läuft, dann die änderungen oder updates der steuerung(en) zurückverfolgen.

wenn das bare metal web interface alleine, dann auch bei beiden wr's gleichzeitig kommunikationsfehler zeigt, dann verschiedene nrf24+ sendeleistungseinstellungen ausprobieren (sowohl hoch als auch runter kann helfen)
und mal einen anderen standort für die opendtu ausprobieren (portable die dtu an eine powerbank stecken und damit spazieren gehen)

welche opendtu version hast du ?
v24.2.16 gilt als sehr stabil, wenn die steuerung/api adapter bereits die api neu unterstützt.

**Comment by JGebauer1** [comment]
Danke für die Ratschläge, leider alles bereits versucht. Das Problem tritt auf, sobald beide WR in openDTU auf Abfrage gesetzt werden, auch wenn sonst keine API-Abfragen laufen. OpenDTU wie gesagt die neueste 24.2.16 generic.
Ich tippe ganz stark auf schwächelnde nrf24, vermutlich sind das einfach billige. Es scheint ja nicht am WR zu liegen, solange nur ein WR von der DTU bedient wird, läuft es problemlos mit Abfrage jede Sekunde. Gelegentlich dauert es mal bis zu 5s, das ist aber kein Problem.
Eine weitere DTU ist bestellt, dieses Mal fest zusammengebaut. Wenn es nicht anders geht, werde ich einfach eine pro Wechselrichter laufen lassen, das ist dann auch kein Drama, solange die sich nicht stören?!

**Comment by JGebauer1** [comment]
Nun ist die neue openDTU da, zeigt aber exakt dasselbe Problem. Komisch, dass es bis vor zwei Tagen alles funktionierte.
Ich habe jetzt jede DTU nur mit einem Wechselrichter verbunden, so geht es auch. Ein wenig umständlich, aber funktioniert. So habe ich auch ein schnelleres Abfrageinterval, praktisch jede Sekunde.

**Comment by tbnobody** [comment]
> praktisch jede Sekunde.

Dir ist aber bewusst, dass ein zu kurzes Interval den ESP so auslastet das er keine anderen Dinge (wie Webinterface etc) mehr abarbeitet?

**Comment by JGebauer1** [comment]
Funktioniert aber gerade super. Das Interval ist schon seit Monaten auf 1s eingestellt, bisher funktionierte das ohne Probleme mit zwei Wechselrichtern. Seit zwei Tagen nicht mehr, nur noch mit einem. Warum ist mir nicht klar. Kann durchaus sein, dass ich momentan den ESP zu sehr ausgelastet habe, aber es müsste doch dann irgendwann wieder so gehen, wie vorher? 
Das Webinterface ist mir ehrlich gesagt gar nicht so wichtig, ich frage die OpenDTU direkt mit dem entsprechenden Adapter in ioBroker ab. (MQTT ist übrigens nicht aktiv)
Es geht jetzt mit den zwei openDTUs, so wird es erst einmal laufen.
Falls aber jemand einen Tipp hat, wie ich das wieder auf eine DTU beschränken kann und trotzdem zwei Wechselrichter zuverlässig auslesen und steuern, das wäre natürlich super...

**Comment by prorun26** [comment]
Hallo in die Runde, 
ich habe exakt das gleiche Kommunikationsproblem mit 2 Wechselrichtern (HMS-800 & HMS-1600).
Ich habe zwei andere DTUs und zwei andere Wechselrichter probiert (anderer HMS-800 & HMS-2000) leider ohne Veränderung.

Wenn ich zwei DTUs verwende stören sich beide gegenseitig. In der Console erkannt man, dass die Abfrage der einen DTU dafür sorgt, dass die Datenpakete der anderen DTU kaputt gehen. Mal kommen die Werte bei DTU1 an mal bei DTU2 und nur ganz selten beide parallel.
![image](https://github.com/user-attachments/assets/5a86228c-0eb4-40fa-bc74-8461c818fbc7)
![image](https://github.com/user-attachments/assets/5d968601-c328-4ca1-aacf-d7a973ae84a0)
![image](https://github.com/user-attachments/assets/de739e87-fddd-4e08-a4de-c365b8542625)

 
Wenn ich nur eine DTU verwende, scheint es so als bricht die Kommunikation zu einem der beiden WRs ab, sobald die Gesamtleistung steigt.
![image](https://github.com/user-attachments/assets/7554e864-d844-4daf-99b6-f0e88afeec2a)
![image](https://github.com/user-attachments/assets/5c2c52c0-4f26-4cca-b3d1-7a4360b145c5)
![image](https://github.com/user-attachments/assets/f1ce9663-0cea-44f9-b31f-7265630a8d6d)

Wenn ich nur eine DTU verwende, und die Leistung beider WRs ist sehr niedrig funktioniert komischerweise alles einwandfrei.
![image](https://github.com/user-attachments/assets/57aacfa1-b3d5-47d9-9a72-b50fda246f38)

Mein Abfrageintervall steht auf 5s und in den Diagrammen sind die Linien unterbrochen wenn keine Kommunikation seit mehr als 30s stattgefunden hat.

**Reply by OttoKarlHeinz** [reply]
Hallo in die Runde,
hatte ein ähnliches Problem mit 2 Wechselrichtern, einer an einer Batterie und der andere an Solarpanels. Mit Abschalten des Wechselrichters an den Solarpanels ging immer die Verbindung der openDTU zum Wechselrichter an der Batterie verloren. Die geschriebenen MQTT Steuerungsbefehle kamen teilweise nach 2 Minuten zum Tragen. Geholfen hat das Schreiben der Sollwerte unter der Bedingung, dass die Wechselrichter nur beschrieben werden, wenn diese WR1/2.reachable = on und WR1/2 producing = on sind.
Siehe auch hier 
https://github.com/tbnobody/OpenDTU/discussions/1946
